static/js/components/CourseList.js

```diff
@@ -1,19 +1,50 @@
 import React from 'react';
 
+import { makeCourseStatusDisplay } from '../util/util';
+
 class CourseList extends React.Component {
   render() {
-    const { courseList } = this.props;
+    const { dashboard, courseList } = this.props;
+    let dashboardLookup = {};
+    for (let course of dashboard.courses) {
+      dashboardLookup[course.id] = course;
+    }
+
+    let tables = courseList.programList.map(program => {
+      let courses = courseList.courseList.
+        filter(course => course.program === program.id).
+        map(course => dashboardLookup[course.id]).
+        filter(course => course);
+
+      return <div key={program.id}>
+        <ul className="course-list-header">
+          <li>{program.title}</li>
+          <li />
```

Comment:
It won't be filled with anything, but we use it as a spaceholder to match the `li` elements in the table below. They're laid out with bourbon-neat which does CSS positioning.